steward/src/cellars/cellar_v2_2.rs

```diff
@@ -0,0 +1,244 @@
+                &cellar_id,
+            );
+            let call = SetStrategistPayoutAddressCall {
+                payout: sp_call_parse_address(params.payout)?,
+            };
+
+            Ok(CellarV2_2Calls::SetStrategistPayoutAddress(call).encode())
+        }
+        Function::SwapPositions(params) => {
+            log_cellar_call(CELLAR_NAME, &SwapPositionsCall::function_name(), &cellar_id);
+            let call = SwapPositionsCall {
+                index_1: params.index_1,
+                index_2: params.index_2,
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2_2Calls::SwapPositions(call).encode())
+        }
+        Function::SetShareLockPeriod(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetShareLockPeriodCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetShareLockPeriodCall {
+                new_lock: string_to_u256(params.new_lock)?,
+            };
+
+            Ok(CellarV2_2Calls::SetShareLockPeriod(call).encode())
+        }
+        // This will ultimately need to be a governance function, but for Seven Sea's live testing we are keeping
+        // it here until they get a feel for what an appropriate value is.
+        Function::SetRebalanceDeviation(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetRebalanceDeviationCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetRebalanceDeviationCall {
+                new_deviation: string_to_u256(params.new_deviation)?,
+            };
+
+            Ok(CellarV2_2Calls::SetRebalanceDeviation(call).encode())
+        }
+        // Function::SetupAdaptor(params) => {
+        //     let adaptor_address = normalize_address(params.adaptor.clone());
+        //     if !ALLOWED_SETUP_ADAPTORS.contains(&adaptor_address.as_str()) {
+        //         return Err(ErrorKind::SPCallError
+        //             .context(format!("adaptor setup not allowed: {}", params.adaptor))
+        //             .into());
+        //     }
+
+        //     log_cellar_call(CELLAR_NAME, &SetupAdaptorCall::function_name(), &cellar_id);
+        //     let call = SetupAdaptorCall {
+        //         adaptor: sp_call_parse_address(params.adaptor)?,
+        //     };
+
+        //     Ok(CellarV2_2Calls::SetupAdaptor(call).encode())
+        // }
+        Function::InitiateShutdown(_) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &InitiateShutdownCall::function_name(),
+                &cellar_id,
+            );
+            let call = InitiateShutdownCall {};
+
+            Ok(CellarV2_2Calls::InitiateShutdown(call).encode())
+        }
+    }
+}
+
+/// Encodes calls to the Adaptor contracts
+fn get_encoded_adaptor_call(data: Vec<AdaptorCall>) -> Result<Vec<AbiAdaptorCall>, Error> {
```

Comment:
This function is exactly the same in both cellar v2 contexts, right? Can we split it out?

Reply:
This is one of those duplications that the build script PR will fix. 

The different cellar v2 definitions have their own AdaptorCall types, requiring two of these functions. The build PR makes the ABI definitions part of the steward crate, which would allow me to impl `From<T>` between the types and have a single function.